Invoice validation #87

Aleksalje opened the pull request:
Invoice validation #87

MilosPaunovic commented on src/components/invoices/Dialog.vue:
```diff
@@ -65,7 +61,9 @@
             lazy-rules
-            :rules="[(val) => val > 0 || 'Amount is must be bigger than zero']"
+            :rules="[
+              (val) => isValidAmount || 'Amount is must be bigger than zero',
```
Check copy for error message.

Aleksalje replied:
Copy corrected

MilosPaunovic commented on src/components/invoices/Dialog.vue:
```diff
@@ -83,17 +81,19 @@
             lazy-rules
-            :rules="[(val) => val > 0 || 'Amount is must be bigger than zero']"
+            :rules="[
+              (val) => isValidExchange || 'Amount is must be bigger than zero',
```
Check copy for error message.

Aleksalje replied:
Copy corrected

MilosPaunovic reviewed (changes requested):
Nice direction, @Aleksalje!

I just think its better, performance wise, to wrap inputs in a`QForm` component than to create separate computed properties for every field.

That way you could revert rules to be field specific, you could leave the `Add` button always enabled but inside a method that handles adding or editing, to validate form first, and in case of an error, mark invalid fields and focus the first errored input from the top.

MilosPaunovic commented:
@Aleksalje @jelenamilanovic 

Take a look at this [document](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) when you find the time, specifically, the PR keywords which will help us automate closing issues.

Aleksalje commented:
Reverted computed rules for fields. 
The issue was that the button was emitting an event to close the popup.
Removed close-popup event and changed the button type to submit.
